commit 688c78af45dc197ce0ef8a0ea520d77592905394
Author: Sebastian Zivota <loewenheim@mailbox.org>
Date:   Wed Apr 16 18:43:43 2025 +0200

    Apply patch

diff --git a/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h b/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
index 3e029a9b..da18a09a 100644
--- a/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
+++ b/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
@@ -58,6 +58,7 @@ struct DemangleOptions {
   bool ShortenArchetype = false;
   bool ShowPrivateDiscriminators = true;
   bool ShowFunctionArgumentTypes = true;
+  bool ShowFunctionReturnType = true;
   bool DisplayDebuggerGeneratedModule = true;
   bool DisplayStdlibModule = true;
   bool DisplayObjCModule = true;
@@ -90,6 +91,7 @@ struct DemangleOptions {
     Opt.ShortenArchetype = true;
     Opt.ShowPrivateDiscriminators = false;
     Opt.ShowFunctionArgumentTypes = false;
+    Opt.ShowFunctionReturnType = false;
     Opt.ShowAsyncResumePartial = false;
     return Opt;
   };
diff --git a/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp b/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
index d9098192..5e8ae258 100644
--- a/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
+++ b/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
@@ -966,10 +966,11 @@ private:
     if (isSendable)
       Printer << "@Sendable ";
 
-    printFunctionParameters(LabelList, node->getChild(argIndex), depth,
-                            Options.ShowFunctionArgumentTypes);
+    if (Options.ShowFunctionArgumentTypes) {
+      printFunctionParameters(LabelList, node->getChild(argIndex), depth, true);
+    }
 
-    if (!Options.ShowFunctionArgumentTypes)
+    if (!Options.ShowFunctionReturnType)
       return;
 
     if (isAsync)
