commit af26ebbf5f3bb03c92739b19127f5c2c5b402e4a
Author: Arpad Borsos <arpad.borsos@sentry.io>
Date:   Mon Oct 19 13:28:23 2020 +0200

    feat(demangle): Add option to skip Swift function parameters

diff --git a/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h b/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
index ce3a4b5..df88a9c 100644
--- a/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
+++ b/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
@@ -57,6 +57,7 @@ struct DemangleOptions {
   bool ShortenArchetype = false;
   bool ShowPrivateDiscriminators = true;
   bool ShowFunctionArgumentTypes = true;
+  bool ShowFunctionArguments = true;
   std::function<std::string(uint64_t, uint64_t)> GenericParameterName =
       genericParameterName;
 
diff --git a/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp b/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
index 1e39291..0784608 100644
--- a/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
+++ b/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
@@ -738,6 +738,11 @@ private:
       setInvalid();
       return;
     }
+
+    if (!Options.ShowFunctionArguments) {
+      return;
+    }
+
     unsigned startIndex = 0;
     if (node->getChild(0)->getKind() == Node::Kind::ThrowsAnnotation)
       startIndex = 1;
