commit 43fca7dd2617ac93f338b5257a2e57c43dcb8154
Author: Sebastian Zivota <loewenheim@mailbox.org>
Date:   Thu Dec 2 16:15:35 2021 +0100

    Apply patch

diff --git a/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h b/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
index db32dbd..f48e1c2 100644
--- a/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
+++ b/symbolic-demangle/vendor/swift/include/swift/Demangling/Demangle.h
@@ -59,6 +59,7 @@ struct DemangleOptions {
   bool ShortenArchetype = false;
   bool ShowPrivateDiscriminators = true;
   bool ShowFunctionArgumentTypes = true;
+  bool ShowFunctionReturnType = true;
   bool DisplayDebuggerGeneratedModule = true;
   bool DisplayStdlibModule = true;
   bool DisplayObjCModule = true;
@@ -90,6 +91,7 @@ struct DemangleOptions {
     Opt.ShortenArchetype = true;
     Opt.ShowPrivateDiscriminators = false;
     Opt.ShowFunctionArgumentTypes = false;
+    Opt.ShowFunctionReturnType = false;
     return Opt;
   };
 };
diff --git a/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp b/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
index 2a9c0dc..34fa785 100644
--- a/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
+++ b/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
@@ -863,10 +863,11 @@ private:
     if (isSendable)
       Printer << "@Sendable ";
 
-    printFunctionParameters(LabelList, node->getChild(startIndex),
-                            Options.ShowFunctionArgumentTypes);
+    if (Options.ShowFunctionArgumentTypes) {
+      printFunctionParameters(LabelList, node->getChild(startIndex), true);
+    }
 
-    if (!Options.ShowFunctionArgumentTypes)
+    if (!Options.ShowFunctionReturnType)
       return;
 
     if (isAsync)
