diff --git a/symbolic-demangle/vendor/swift/include/assert.h b/symbolic-demangle/vendor/swift/include/assert.h
new file mode 100644
index 00000000..e981921a
--- /dev/null
+++ b/symbolic-demangle/vendor/swift/include/assert.h
@@ -0,0 +1,12 @@
+// Our overwrite of assert, this is done so that we get an exception which can be caught before the
+// FFI, rather than aborting.
+#include <string>
+#include <stdexcept>
+
+#undef assert
+#define assert(cond)                                                          \
+  do {                                                                        \
+    if (!(cond))                                                              \
+      throw std::logic_error("Assertion failed: (" #cond "), file "           \
+                              __FILE__ ", line " + std::to_string(__LINE__)); \
+  } while (0)
diff --git a/symbolic-demangle/vendor/swift/lib/Demangling/DemanglerAssert.h b/symbolic-demangle/vendor/swift/lib/Demangling/DemanglerAssert.h
index 367a81ef..a734aa36 100644
--- a/symbolic-demangle/vendor/swift/lib/Demangling/DemanglerAssert.h
+++ b/symbolic-demangle/vendor/swift/lib/Demangling/DemanglerAssert.h
@@ -22,7 +22,6 @@
 #include "swift/Demangling/Demangle.h"
 #include "swift/Demangling/NamespaceMacros.h"

-#if defined(NDEBUG) || defined (SWIFT_RUNTIME)

 // In the runtime and non-asserts builds, DEMANGLER_ASSERT() returns an error.
 #define DEMANGLER_ASSERT(expr, node)                                           \
@@ -31,21 +30,6 @@
       return ManglingError(ManglingError::AssertionFailed, (node), __LINE__);  \
   } while (0)

-#else
-
-// Except in unittests, assert builds cause DEMANGLER_ASSERT() to assert()
-#define DEMANGLER_ASSERT(expr, node)                                           \
-  do {                                                                         \
-    if (!(expr)) {                                                             \
-      if (Factory.disableAssertionsForUnitTest)                                \
-        return ManglingError(ManglingError::AssertionFailed, (node),           \
-                             __LINE__);                                        \
-      else                                                                     \
-        swift::Demangle::failAssert(__FILE__, __LINE__, node, #expr);          \
-    }                                                                          \
-  } while (0)
-
-#endif

 // DEMANGLER_ALWAYS_ASSERT() *always* fails the program, even in the runtime
 #define DEMANGLER_ALWAYS_ASSERT(expr, node)                                    \
diff --git a/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp b/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
index 7431e548..6e60bee7 100644
--- a/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
+++ b/symbolic-demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
@@ -27,6 +27,7 @@ using namespace swift;
 using namespace Demangle;
 using llvm::StringRef;

+
 DemanglerPrinter &DemanglerPrinter::operator<<(unsigned long long n) & {
   char buffer[32];
   snprintf(buffer, sizeof(buffer), "%llu", n);
@@ -50,8 +51,7 @@ DemanglerPrinter &DemanglerPrinter::operator<<(long long n) & {

 [[noreturn]]
 static void printer_unreachable(const char *Message) {
-  fprintf(stderr, "fatal error: %s\n", Message);
-  std::abort();
+  throw std::logic_error(Message);
 }

 std::string Demangle::genericParameterName(uint64_t depth, uint64_t index) {
@@ -2889,7 +2889,7 @@ NodePointer NodePrinter::print(NodePointer Node, unsigned depth,
     return nullptr;
   case Node::Kind::ImplErasedIsolation:
     Printer << "@isolated(any)";
-    return nullptr;
+    return nullptr;
   case Node::Kind::ImplCoroutineKind:
     // Skip if text is empty.
     if (Node->getText().empty())
@@ -2984,7 +2984,7 @@ NodePointer NodePrinter::print(NodePointer Node, unsigned depth,
   case Node::Kind::ErrorType:
     Printer << "<ERROR TYPE>";
     return nullptr;
-
+
   case Node::Kind::DependentPseudogenericSignature:
   case Node::Kind::DependentGenericSignature: {
     printGenericSignature(Node, depth);
@@ -3605,10 +3605,7 @@ NodePointer NodePrinter::printEntity(NodePointer Entity, unsigned depth,
   if (TypePr != TypePrinting::NoType) {
     NodePointer type = getChildIf(Entity, Node::Kind::Type);
     assert(type && "malformed entity");
-    if (!type) {
-      setInvalid();
-      return nullptr;
-    }
+
     type = type->getChild(0);
     if (TypePr == TypePrinting::FunctionStyle) {
       // We expect to see a function type here, but if we don't, use the colon.
