commit 575a6035e76e73eacc99d43a412d0afa2068dd25
Author: Jan Michael Auer <mail@jauer.org>
Date:   Sun Apr 5 16:39:15 2020 +0200

    feat(demangle): Add option to skip Swift function parameters

diff --git a/demangle/vendor/swift/include/swift/Demangling/Demangle.h b/demangle/vendor/swift/include/swift/Demangling/Demangle.h
index 2a8fd6c..c377a00 100644
--- a/demangle/vendor/swift/include/swift/Demangling/Demangle.h
+++ b/demangle/vendor/swift/include/swift/Demangling/Demangle.h
@@ -57,6 +57,7 @@ struct DemangleOptions {
   bool ShortenArchetype = false;
   bool ShowPrivateDiscriminators = true;
   bool ShowFunctionArgumentTypes = true;
+  bool ShowFunctionArguments = true;
   std::function<std::string(uint64_t, uint64_t)> GenericParameterName =
       genericParameterName;
 
diff --git a/demangle/vendor/swift/lib/Demangling/NodePrinter.cpp b/demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
index 097be7f..266506e 100644
--- a/demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
+++ b/demangle/vendor/swift/lib/Demangling/NodePrinter.cpp
@@ -728,6 +728,11 @@ private:
       setInvalid();
       return;
     }
+
+    if (!Options.ShowFunctionArguments) {
+      return;
+    }
+
     unsigned startIndex = 0;
     if (node->getChild(0)->getKind() == Node::Kind::ThrowsAnnotation)
       startIndex = 1;
